Kauri/3box-comments-tutorial.md

```diff
@@ -0,0 +1,333 @@
+
+![React app setup screenshot]()
+
+### Importing 3Box
+
+Now we're going to start setting up our `3box` instance. We'll import `3box` and `3box-comments-react` at the top of `src/App.js` file:
+
+```js
+// src/App.js
+import React from 'react';
+import Box from '3box';
+import BoxComments from '3box-comments-react';
+
+...
+```
+
+### Adding Component State
+
+Our `App` component will need to [store some state](https://reactjs.org/docs/state-and-lifecycle.html#adding-local-state-to-a-class) for later. Particularly, we'll need to store a `3box` instance, the user's Ethereum address, the user's 3Box profile (if any), and a readiness flag:
+
+```js
+// src/App.js
+...
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      box: {},
+      profile: {},
+      address: '',
+      ready: false,
+    };
+  }
+
+  ...
+}
+
+...
+```
+
+### Initializing 3Box
+
+We're ready to initialize 3Box. We're going create our `3box` instance after the component is mounted by listening for the [`componentDidMount`](https://reactjs.org/docs/react-component.html#componentdidmount) lifecycle hook. We'll need to create a function `login()` that actually handles the `3box` initialization logic.
+
+Note here that we need access to an `ethereum` instance in order to initialize `3box`. Again, you can create an `ethereum` instance with a library like Web3, but our browser wallet will automatically inject an `etherum` instance on our behalf.
+
+Our `login()` function is going to pull the user's address from `ethereum` and then use that address to connect to `3box`. We'll be ready to create the component once this first step is complete. Our `3box` instance will do some light synchronization in the background, so we'll also update the component once synchronization has finished.
+
+Adding these new functions:
+
+```js
+// src/App.js
+...
+
+class App extends React.Component {
+  ...
+
+  componentDidMount() {
+    this.login();
+  }
+
+  async login() {
+    // Request access to wallet and pull addresses.
+    const addresses = await window.ethereum.enable();
+    const address = addresses[0];
+
+    // Create the 3Box instance.
+    const box = await Box.openBox(address, window.ethereum, {});
+    const profile = await Box.getProfile(address);
+
+    // Update the box in our state once we've synced.
+    box.onSyncDone(() => {
+      this.setState({
+        box: box
+      });
+    });
+
+    // Update our state immediately with current values.
+    this.setState({
+      box: box,
+      profile: profile,
+      address: address,
+      ready: true,
+    });
+  }
+
+  ...
+}
+
+...
+```
+
+### Creating the Component
+
+We're finally ready to create our component. We're going to update our [`render()`](https://reactjs.org/docs/react-component.html#render) function so that it renders the component we imported from `3box-comments-react`. Once we're finished with this step, we'll have have a working comment section!
```

Comment:
```suggestion
We're finally ready to create our component. We update our [`render()`](https://reactjs.org/docs/react-component.html#render) function so that it renders the component we imported from `3box-comments-react`. Once we're finished with this step, we have have a working comment section!
```